worker-document-schema/src/main/resources-yaml/com/microfocus/caf/worker/document/schema/model/schema.yaml

```diff
@@ -0,0 +1,167 @@
+#
+# Copyright 2016-2019 Micro Focus or one of its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"$schema": http://json-schema.org/draft-07/schema#
+"$id": https://cafdataprocessing.github.io/worker-document/schema.json
+title: Document worker wocument schema
+description: The schema for a document that a Document Worker can process.
+type: object   
+properties: 
+  document:
+    "$ref": "#/definitions/workerDocument"
+  changeLog:
+    "$ref": "#/definitions/changeLog"
+additionalProperties: false
+definitions:
+  workerDocument:
+    type: object
+    properties:
+      reference:
+        type: string
+      fields:
+        type: object
+        additionalProperties:
+          type: array
+          items:
+            "$ref": "#/definitions/fieldValue"
+      failures:
+        type: array
+        items:
+          type: object
+          "$ref": "#/definitions/failure"
+      subdocuments:
+        type: array
+        items:
+          type: object
+          "$ref": "#/definitions/workerDocument"
+    additionalProperties: false
+  fieldValue:
+    type: object
+    required:
+    - data
+    properties:
+      data:
+        type: string
+      encoding:
+        type: string
+        default: utf8
+        enum: [utf8, base64, storage_ref, local_ref]
+    additionalProperties: false
+  failure:
+    type: object
+    properties:
+      failureId:
+        type: string
+      failureMessage:
+        type: string
+      failureStack:
+        type: string
+    additionalProperties: false
+  changeLog:
+    type: array
+    items:
+      type: object
+      "$ref": "#/definitions/changeLogEntry"
+  changeLogEntry:
+    type: object
+    properties:
+      name:
+        type: string
+      changes:
+        type: array
+        items:
+          type: object
+          "$ref": "#/definitions/change"
+    additionalProperties: false
+  change:
+    type: object
+    oneOf:
+    - required: [setReference]
+    - required: [addFields]
+    - required: [setFields]
+    - required: [addFailure]
+    - required: [addSubdocument]
```

Comment:
I think `setFailures` is missing from this list.

Reply:
added setFailures